code/modules/chemistry/Reagents-FoodDrink.dm

```diff
@@ -4573,6 +4573,17 @@ datum
 					boutput(M, "<font color=#be9ffe>You feel [.].</font>")
 
 				..()
+			on_add()
+				if(holder.get_reagent_amount("miasma") > 0)
+					var/lavenderunits = ((holder.get_reagent_amount("lavender_essence")*2) - holder.get_reagent_amount("miasma"))
+					if(lavenderunits >= 0)
+						holder.del_reagent("miasma")
+						holder.del_reagent("lavender_essence")
+						holder.add_reagent("lavender_essence", (lavenderunits/2))
+					if(lavenderunits < 0)
+						holder.del_reagent("miasma")
+						holder.del_reagent("lavender_essence")
+						holder.add_reagent("miasma", (abs(lavenderunits)))
```

Comment:
```suggestion
var/miasma_amount = src.holder.get_reagent_amount("miasma")
src.holder.remove_reagent("miasma", src.get_reagent_amount(src.id) * 2)
src.holder.remove_reagent(src.id, miasma_amount/2)
```
This seems a lot clearer than the above (and I think does the same thing?)

Reply:
Okay thank you the reason why the code was like that is because I knew that there was a way to get the code down to three lines but I am a dumb baby and overthought it.